src/test/scala/uk/gov/nationalarchives/omega/jms/JmsRRClientSpec.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2023 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.jms
+
+import cats.effect.std.Queue
+import cats.effect.{IO, Resource}
+import munit.CatsEffectSuite
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+import org.typelevel.log4cats.slf4j.Slf4jFactory
+import uk.gov.nationalarchives.omega.jms.JmsRRClient.ReplyMessageHandler
+
+class JmsRRClientSpec extends CatsEffectSuite {
+
+  test("request reply test") {
```

Comment:
This test never completes for me.
I have run the docker compose file, as well as the stub. I can see the stub's logs:
```
Echo Server received message: hello 1234
Echo Server sending response message: Echo Server: hello 1234 with correlationId: ID:0deba559-44bc-4eae-82c5-5078d4921906
[io-compute-blocker-9] INFO com.amazon.sqs.javamessaging.SQSMessageProducer - Message sent to SQS with SQS-assigned messageId: 00fc7c99-3f9c-4e68-893b-0c1b4c1b7e13
```
But the test never seems to acknowledge the receipt of the reply from the stub. I've spent a bit of time looking at this, and I don't think the reply handler is ever executed.

Reply:
After restarting the stub and queue several times the test passes seemingly consistently for me. I've made no changes, but now I'm unable to reproduce the issue above. (note that before I was waiting quite a long time for the test to complete, upwards of 10 minutes).

Reply:
As mentioned [here](https://national-archives.atlassian.net/browse/PACT-685?focusedCommentId=39953), I've only had one success.

Then again, I only waited 2 minutes, tops.

In any case, it will need to be quite quick to be usable.